libtransmission/web.cc

```diff
@@ -409,14 +409,21 @@ class tr_web::Impl
         return version >= 0x075700 /* 7.87.0 */ && version <= 0x080500 /* 8.5.0 */;
     }
 
+    // https://github.com/curl/curl/issues/6312
+    [[nodiscard]] static bool check_curl_gh6312() noexcept
+    {
+        auto const version = curl_version_info(CURLVERSION_NOW)->version_num;
```

Comment:
On a side note, it looks like we call this in several places, e.g. here + `check_curl_gh10936()` + `ssl_context_func()`.

Is this value ever going to change during a session? Since it's looking at the version of curl that was loaded when the process started, it seems like it wouldn't change and we can cache it?

Not a new issue / not a blocker for this PR, but looks like it might be a small followup perf refactor that we could do e.g.

```
[[nodiscard]] static auto get_curl_version_num() noexcept
{
    static const auto version_num = curl_version_info(CURLVERSION_NOW)->version_num;
    return version_num;
}

Reply:
Yes, definitely a follow-up PR here. It can even be directly written in the member initializer:

```c++
bool const curl_version_num = curl_version_info(CURLVERSION_NOW)->version_num;
```

Reply:
followup @ https://github.com/transmission/transmission/pull/7461